x/stakeibc/keeper/msg_server_create_trade_route_test.go

```diff
@@ -0,0 +1,231 @@
+
+	// Create a host zone with an exisiting withdrawal address
+	hostZone := types.HostZone{
+		ChainId:              HostChainId,
+		ConnectionId:         hostConnectionId,
+		WithdrawalIcaAddress: withdrawalAddress,
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	// Define a valid message given the parameters above
+	msg = types.MsgCreateTradeRoute{
+		Authority:   Authority,
+		HostChainId: HostChainId,
+
+		StrideToRewardConnectionId: rewardConnectionId,
+		StrideToTradeConnectionId:  tradeConnectionId,
+
+		HostToRewardTransferChannelId:  hostToRewardChannelId,
+		RewardToTradeTransferChannelId: rewardToTradeChannelId,
+		TradeToHostTransferChannelId:   tradeToHostChannelId,
+
+		RewardDenomOnHost:   rewardDenomOnHost,
+		RewardDenomOnReward: rewardDenomOnReward,
+		RewardDenomOnTrade:  rewardDenomOnTrade,
+		HostDenomOnTrade:    hostDenomOnTrade,
+		HostDenomOnHost:     hostDenomOnHost,
+
+		PoolId:                 poolId,
+		MaxAllowedSwapLossRate: maxAllowedSwapLossRate,
+		MinSwapAmount:          minSwapAmount,
+		MaxSwapAmount:          maxSwapAmount,
+	}
+
+	// Build out the expected trade route given the above
+	expectedTradeRoute = types.TradeRoute{
+		RewardDenomOnHostZone:   rewardDenomOnHost,
+		RewardDenomOnRewardZone: rewardDenomOnReward,
+		RewardDenomOnTradeZone:  rewardDenomOnTrade,
+		HostDenomOnTradeZone:    hostDenomOnTrade,
+		HostDenomOnHostZone:     hostDenomOnHost,
+
+		HostAccount: types.ICAAccount{
+			ChainId:      HostChainId,
+			Type:         types.ICAAccountType_WITHDRAWAL,
+			ConnectionId: hostConnectionId,
+			Address:      withdrawalAddress,
+		},
+		RewardAccount: types.ICAAccount{
+			ChainId:      rewardChainId,
+			Type:         types.ICAAccountType_CONVERTER_UNWIND,
+			ConnectionId: rewardConnectionId,
+			Address:      unwindAddress,
+		},
+		TradeAccount: types.ICAAccount{
+			ChainId:      tradeChainId,
+			Type:         types.ICAAccountType_CONVERTER_TRADE,
+			ConnectionId: tradeConnectionId,
+		},
+
+		HostToRewardChannelId:  hostToRewardChannelId,
+		RewardToTradeChannelId: rewardToTradeChannelId,
+		TradeToHostChannelId:   tradeToHostChannelId,
+
+		TradeConfig: types.TradeConfig{
+			PoolId:               poolId,
+			SwapPrice:            sdk.ZeroDec(),
+			PriceUpdateTimestamp: 0,
+
+			MaxAllowedSwapLossRate: sdk.MustNewDecFromStr(maxAllowedSwapLossRate),
+			MinSwapAmount:          minSwapAmount,
+			MaxSwapAmount:          maxSwapAmount,
+		},
+	}
+
+	return msg, expectedTradeRoute
+}
+
+// Helper function to create a trade route and check the created route matched expectations
+func (s *KeeperTestSuite) submitCreateTradeRouteAndValidate(msg types.MsgCreateTradeRoute, expectedRoute types.TradeRoute) {
+	_, err := s.GetMsgServer().CreateTradeRoute(sdk.WrapSDKContext(s.Ctx), &msg)
+	s.Require().NoError(err, "no error expected when creating trade route")
+
+	actualRoute, found := s.App.StakeibcKeeper.GetTradeRoute(s.Ctx, RewardDenom, HostDenom)
```

Comment:
Should you be loading these denoms off the expectedRoute rather than having them hard coded as globvars from the above setup?

Reply:
Either is fine imo. I think anytime we have only one trade route in a test, we can just use the global variables